paper-linear-progress.html

```diff
@@ -0,0 +1,897 @@
+         * representing the percentage complete of buffering. Note that the value will be clamped
+         * if it is less than/greater than 0/100, respectively.
+         */
+        bufferValue: {
+          type: Number,
+          observer: '_bufferValueChanged',
+          value: 100
+        },
+
+        /**
+         * When set, progress bars animate across the element, expressing an operation which takes
+         * an unknown amount of time.
+         */
+        indeterminate: {
+          type: Boolean,
+          observer: '_indeterminateChanged',
+          // We give a default value so that correct ARIA attributes are applied when element is
+          // initialized.
+          value: false
+        },
+
+        /**
+         * When set, will reverse all animations, transforms, and transitions occuring within the
+         * element. Useful when working with RTL languages, as well as distinguishing between phases
+         * of an operation.
+         */
+        reversed: {
+          type: Boolean,
+          observer: '_reversedChanged'
+        },
+
+        /**
+         * When set, will display buffering dots which appear to animate across the element. Useful
+         * when displaying progress for operations involving media, during which content may need to
+         * be buffered.
+         */
+        buffering: {
+          type: Boolean,
+          observer: '_bufferingChanged'
+        },
+
+        /**
+         * When set, the linear progress element will be hidden from view. Note that this is
+         * _different_ from `hidden`, as it does not set `display: none;` but rather just changes
+         * the element's opacity.
+         */
+        closed: {
+          type: Boolean,
+          observer: '_closedChanged'
+        },
+
+        /**
+         * Animation configuration for NeonAnimationRunnerBehavior.
+         */
+        animationConfig: {
+          value: function() {
+            // NOTE(traviskaufman): This is initialized within the attached() method, once we know
+            // the height of the progress indicator.
+            return {};
+          }
+        },
+
+        __needsBufferingAnimationOnAttached: {
+          // HACK: We want to have buffering play an animation when it's set, but we can't
+          // animate unless we first have the computed height of the element, which we can only
+          // retrieve once the element's attached. Therefore, we set up a queue of animations that
+          // need to be run, and if we encounter any animations. Note that this can be fixed once
+          // https://github.com/Polymer/polymer/issues/2399 is resolved. Then, we can do all
+          // animations within CSS.
+          type: Boolean,
+          value: false
+        }
+      },
+
+      hostAttributes: {
+        role: 'progressbar'
+      },
+
+      ready: function() {
+        this.toggleClass('isIE', /MSIE/.test(navigator.userAgent));
```

Comment:
We strive never to write code based on user-agent string matching. Granted, the difficulty of some UA-specific problems becomes asymptotic. That said, could you describe the IE-specific problem in detail so that we can collaborate on a solution that does not rely on user-agent string matching?